psl-core/src/main/java/org/linqs/psl/application/learning/weight/bayesian/GaussianProcessPrior.java

```diff
@@ -317,11 +399,16 @@ protected float getFunctionValue(WeightConfig config) {
     protected int getNextPoint(List<WeightConfig> configs, int iteration) {
         int bestConfig = -1;
         float curBestVal = -Float.MAX_VALUE;
-        for (int i = 0; i < configs.size(); i++) {
-            float curVal = (configs.get(i).valueAndStd.value / exploration) + configs.get(i).valueAndStd.std;
-            if (bestConfig == -1 || curVal > curBestVal) {
-                curBestVal = curVal;
-                bestConfig = i;
+
+        if ((iteration == 0) && useProvidedWeight) {
+            bestConfig = PROVIDED_CONFIG_INDEX;
```

Comment:
Just return right away, and then the rest doesn't need to be in an additional block.